cumulus/ansible/tasks/providers/base.py

```diff
@@ -0,0 +1,48 @@
+class InstanceState():
+    PENDING = "pending"
+    RUNNING = "running"
+    SHUTTINGDOWN = "shutting-down"
+    TERMINATED = "terminated"
+    STOPPING = "stopping"
+    STOPPED = "stopped"
+
+
+class Provider(object):
+    __provider_registry__ = {}
```

Comment:
CloudProvider sounds fine to me. The purpose of the class(es) is to provide a consistent read-only interface to cloud providers like AWS. This lets us check state,  fail early and build/modify ansible inventories